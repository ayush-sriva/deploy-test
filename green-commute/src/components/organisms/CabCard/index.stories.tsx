import { ComponentMeta, ComponentStory } from '@storybook/react';
import CabCard from '.';
import Olaurl from '../../../../public/assets/logos/ola.png';
import Rapido from '../../../../public/assets/logos/rapido.png';
import Uber from '../../../../public/assets/logos/uber.png';

export default {
  title: 'components/Organisms/CabCard',
  component: CabCard,
} as ComponentMeta<typeof CabCard>;
const data = [
  {
    id: 'CabData1001',
    logoUrl: `${Olaurl}`,
    company: 'Ola',
    approx: 'Approximately  ',
    fare: '45',
  },
  {
    id: 'CabData1002',
    logoUrl: `${Rapido}`,
    company: 'Ola',
    approx: 'Approximately  ',
    fare: '45',
  },
  {
    id: 'CabData1003',
    logoUrl: `${Uber}`,
    company: 'Ola',
    approx: 'Approximately  ',
    fare: '45',
  },
];

const Template: ComponentStory<typeof CabCard> = (args) => <CabCard {...args} />;

export const CabCardComponent = Template.bind({});

CabCardComponent.args = {
  data: data,
};
