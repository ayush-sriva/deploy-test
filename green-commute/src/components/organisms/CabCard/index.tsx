/**
 * @author Asish Kalintha <asish.sugun@zemosolabs.com>
 */

import { Grid, styled } from '@mui/material';
import TypographyComponent from '../../atoms/Typography';
import responsiveTheme from '../../../theme';
import Eclipse from '../../../../public/assets/icons/eclipse.svg';
import Icon from '../../atoms/Icon';
import ListItem from '../../molecules/ListItem';

interface CabData {
  id: string;
  logoUrl: string;
  company: string;
  approx: string;
  fare: string;
}
interface CabCardProps {
  data: Array<CabData>;
}

const GridMainContainer = styled(Grid)({
  width: '25.2rem',
  height: '18rem',
  padding: '1.3rem',
});

const GridtypographyItem = styled(Grid)({
  color: `${responsiveTheme.palette.secondary.dark}`,
  paddingBottom: '0.3rem',
});

const GridSubContainer = styled(Grid)({
  color: `${responsiveTheme.palette.secondary.darker}`,
  paddingBottom: '0.5rem',
});

const GridIconItem = styled(Grid)({
  paddingLeft: '0.6rem',
  paddingRight: '0.6rem',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});
const GridBoxWithPadding = styled(Grid)({
  paddingBottom: '0rem',
});

const CabCard: React.FC<CabCardProps> = ({ data }) => {
  return (
    <div>
      <GridMainContainer container direction='column'>
        <GridtypographyItem item>
          <TypographyComponent variant={'body2'} children={'catch a blue line metro towards Raidurg.'} />
        </GridtypographyItem>
        <Grid item>
          <GridSubContainer container>
            <Grid item>
              <TypographyComponent variant={'caption'} children={'â‚¹ 100'} />
            </Grid>
            <GridIconItem item>
              <Icon url={Eclipse} />
            </GridIconItem>
            <Grid item>
              <TypographyComponent variant={'caption'} children={'25 Kms'} />
            </Grid>
            <GridIconItem item>
              <Icon url={Eclipse} />
            </GridIconItem>
            <Grid item>
              <TypographyComponent variant={'caption'} children={'1 hr 20 min'} />
            </Grid>
          </GridSubContainer>
        </Grid>
        {data.map((currentData) => {
          return (
            <GridBoxWithPadding item key={currentData.id} data-testid={currentData.id}>
              <ListItem
                logoUrl={currentData.logoUrl}
                company={currentData.company}
                approx={currentData.approx}
                fare={currentData.fare}
              />
            </GridBoxWithPadding>
          );
        })}
      </GridMainContainer>
    </div>
  );
};

export default CabCard;
