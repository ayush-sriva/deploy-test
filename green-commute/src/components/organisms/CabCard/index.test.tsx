import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CabCard from '.';
import Olaurl from '../../../../public/assets/logos/ola.png';
import Rapido from '../../../../public/assets/logos/rapido.png';
import Uber from '../../../../public/assets/logos/uber.png';
  
const data = [
    {
      id: 'CabData1001',
      logoUrl: `${Olaurl}`,
      company: 'Ola',
      approx: 'Approximately  ',
      fare: '45',
    },
    {
      id: 'CabData1002',
      logoUrl: `${Rapido}`,
      company: 'Ola',
      approx: 'Approximately  ',
      fare: '45',
    },
    {
      id: 'CabData1003',
      logoUrl: `${Uber}`,
      company: 'Ola',
      approx: 'Approximately  ',
      fare: '45',
    },
  ];

function renderCabCard() {

  render(<CabCard data={data} />);
}

describe('CabCard', () => {

  it('renders correctly', () => {
     const tree = renderer.create(<CabCard data={data} />).toJSON();
       expect(tree).toMatchSnapshot();
  });

  test('Text should render correctly', () => {
    renderCabCard();
    const ButtonElement = screen.getByText(/catch a blue line metro towards Raidurg./);
    expect(ButtonElement).toBeInTheDocument();
  });

  test('Element for ola should be rendered', () => {
    renderCabCard();
    const CabElement = screen.getByTestId('CabData1001');
    expect(CabElement).toBeInTheDocument();
  });
  test('Element rapido should be rendered', () => {
    renderCabCard();
    const CabElement = screen.getByTestId('CabData1002');
    expect(CabElement).toBeInTheDocument();
  });
  test('Element uber should be rendered', () => {
    renderCabCard();
    const CabElement = screen.getByTestId('CabData1003');
    expect(CabElement).toBeInTheDocument();
  });
});
