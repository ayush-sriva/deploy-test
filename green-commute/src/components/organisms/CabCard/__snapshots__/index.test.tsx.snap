// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CabCard renders correctly 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-yg2epo-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-1mezf00-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        catch a blue line metro towards Raidurg.
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-1y7hg28-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <span
            className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            ₹ 100
          </span>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-1t46eol-MuiGrid-root"
        >
          <img
            alt="Icon"
            data-testid="icon"
            src="eclipse.svg"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <span
            className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            25 Kms
          </span>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-1t46eol-MuiGrid-root"
        >
          <img
            alt="Icon"
            data-testid="icon"
            src="eclipse.svg"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <span
            className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            1 hr 20 min
          </span>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-gw4rzq-MuiGrid-root"
      data-testid="CabData1001"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-paper-1 css-1ps6pg7-MuiPaper-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-15hauir-MuiGrid-root"
          data-testid="main-grid"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1.5 makeStyles-grid-2 css-1duidt6-MuiGrid-root"
            data-testid="logo-grid"
          >
            <img
              alt="Logo"
              data-testid="logo"
              src="ola.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 MuiGrid-grid-sm-true css-unluoz-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column css-1oolkk7-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-137v0wv-MuiGrid-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-captionBold css-37jdci-MuiTypography-root"
                >
                  Ola
                </span>
              </div>
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-137v0wv-MuiGrid-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
                >
                  Approximately  
                   ₹
                  45
                </span>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
            data-testid="book-now"
          >
            <span
              className="MuiTypography-root MuiTypography-captionBold makeStyles-colorText-3 css-37jdci-MuiTypography-root"
            >
              Book Now
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-gw4rzq-MuiGrid-root"
      data-testid="CabData1002"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-paper-1 css-1ps6pg7-MuiPaper-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-15hauir-MuiGrid-root"
          data-testid="main-grid"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1.5 makeStyles-grid-2 css-1duidt6-MuiGrid-root"
            data-testid="logo-grid"
          >
            <img
              alt="Logo"
              data-testid="logo"
              src="rapido.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 MuiGrid-grid-sm-true css-unluoz-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column css-1oolkk7-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-137v0wv-MuiGrid-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-captionBold css-37jdci-MuiTypography-root"
                >
                  Ola
                </span>
              </div>
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-137v0wv-MuiGrid-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
                >
                  Approximately  
                   ₹
                  45
                </span>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
            data-testid="book-now"
          >
            <span
              className="MuiTypography-root MuiTypography-captionBold makeStyles-colorText-3 css-37jdci-MuiTypography-root"
            >
              Book Now
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-gw4rzq-MuiGrid-root"
      data-testid="CabData1003"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-paper-1 css-1ps6pg7-MuiPaper-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-15hauir-MuiGrid-root"
          data-testid="main-grid"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1.5 makeStyles-grid-2 css-1duidt6-MuiGrid-root"
            data-testid="logo-grid"
          >
            <img
              alt="Logo"
              data-testid="logo"
              src="uber.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 MuiGrid-grid-sm-true css-unluoz-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column css-1oolkk7-MuiGrid-root"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-137v0wv-MuiGrid-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-captionBold css-37jdci-MuiTypography-root"
                >
                  Ola
                </span>
              </div>
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-137v0wv-MuiGrid-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
                >
                  Approximately  
                   ₹
                  45
                </span>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
            data-testid="book-now"
          >
            <span
              className="MuiTypography-root MuiTypography-captionBold makeStyles-colorText-3 css-37jdci-MuiTypography-root"
            >
              Book Now
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
